### PR TITLE
Add zoom slider

### PR DESCRIPTION
**Pull request by chfzhang**
Slider sets interval to (powers of 2) + 1.
Using the "+", "-", or "Go" buttons sets the slider appropriately to the closest value.

### CODE REVIEW

File: src/main/utils.js [hunk 1]
```diff
@@ -182,6 +182,20 @@ function scaleRange(range: Interval, factor: number): Interval {
   return new Interval(start, stop);
 }
 
+function absScaleRange(range: Interval, level: number, halfSpan: number): Interval {
```
Comment (akmorrow13) [inline src/main/utils.js]: Can you add some comments to what this function does?

File: src/main/Controls.js [hunk 2]
```diff
@@ -86,6 +94,18 @@ class Controls extends React.Component<Props> {
     this.zoomByFactor(2.0);
   }
 
+  zoomAbs(level: number) {
```
Comment (akmorrow13) [inline src/main/Controls.js]: Can you comment what this function does?

File: src/main/Controls.js [hunk 3]
```diff
@@ -24,6 +24,7 @@ class Controls extends React.Component<Props> {
 
   constructor(props: Object) {
     super(props);
+    this.state = {defaultHalfInterval:2};
```
Comment (akmorrow13) [inline src/main/Controls.js]: what is defaultHalfInterval? Can you add a comment here to define what it is?

**Comment by akmorrow13** [comment]
Wow @chfzhang this is amazing! I left a review asking to add some comments to some of the functions, other than that looks great! The [jenkins tests are currently failing ](https://travis-ci.org/hammerlab/pileup.js/builds/522418964?utm_source=github_status&utm_medium=notification). Make sure you run `flow check` to make sure flow tests are passing. Let me know if you have any questions about this!

File: src/main/utils.js (hunk 1 above)
Comment (chfzhang) [reply src/main/utils.js]: Done.

File: src/main/Controls.js (hunk 2 above)
Comment (chfzhang) [reply src/main/Controls.js]: Done.

File: src/main/Controls.js (hunk 3 above)
Comment (chfzhang) [reply src/main/Controls.js]: Done.

File: src/main/utils.js (hunk 1 above)
Comment (akmorrow13) [reply src/main/utils.js]: Thanks @chfzhang ! I pulled and tested it out and it works great. I am going to squash and merge.

**Pull request merged**